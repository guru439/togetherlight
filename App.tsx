/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import type { PropsWithChildren } from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
} from 'react-native';

import {
  Colors,
  DebugInstructions,
  Header,
  LearnMoreLinks,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import LoginScreen from './screens';
import Posts from './screens/Component/Posts';
import PostDetail from './screens/Component/PostDetail';
import { Provider } from 'react-redux';
import store from './redux/store/store'


type SectionProps = PropsWithChildren<{
  title: string;
}>;

const Stack = createStackNavigator()

function App(): JSX.Element {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator initialRouteName="Login" >
          <Stack.Screen
            name="Login" options={{
              title: 'Welcome Back!', headerStyle: {
                backgroundColor: '#fff',
              }, headerTitleStyle: {
                color: '#0D094E',
                fontSize: 30,
                marginTop: 20,
                fontWeight: 'bold'
              }
            }} component={LoginScreen} />
          <Stack.Screen
            name="Posts"
            options={{
              headerStyle: {
                backgroundColor: '#fff',
              }, headerTitleStyle: {
                color: '#0D094E',
                fontSize: 20,
                fontWeight: 'bold',

              }
            }}
            component={Posts} />
          <Stack.Screen
            options={{
              headerStyle: {
                backgroundColor: '#fff',
              }, headerTitleStyle: {
                color: '#0D094E',
                fontSize: 20,
                fontWeight: 'bold',

              }
            }}
            name="PostDetail" component={PostDetail} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
}


export default App;
