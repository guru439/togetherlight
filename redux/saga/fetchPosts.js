import { call, put, takeEvery, takeLatest } from 'redux-saga/effects'
import * as Api from '../API_URI'
import * as actionTypes from "../_ActionsType"
import { getPosts } from '../../ApiCalls/getPosts';

// Worker saga will be fired on POST_FETCH_REQUESTED actions
export function* fetchPosts(action) {
   try {
        console.log(Api.BASE_API + Api.POSTS)
        const posts = yield call(getPosts);
        console.log(posts)
        yield put({type: actionTypes.UPDATE_POSTS_DATA_SUCCESS, payload: posts});
        return posts
   } catch (e) {
    console.log(e)
      yield put({type: actionTypes.UPDATE_POSTS_DATA_FAIL, payload: e.message});
      return e.message
   }
}

// Starts fetchPost on each dispatched POST_FETCH_REQUESTED action
// Allows concurrent fetches of posts
export default function* mySaga() {
  // yield takeEvery(actionTypes.GET_POSTS_START, fetchPosts);
}