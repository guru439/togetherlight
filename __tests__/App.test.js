/**
 * @format
 */

import 'react-native';
// import '@types/jest';
import LoginScreen from '../screens';

// Note: import explicitly to use the types shiped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import { render, screen, fireEvent } from '@testing-library/react-native';
import rendrer from 'react-test-renderer'

import App from '../App';

import {fetchPosts} from '../redux/saga/fetchPosts'
import {runSaga} from 'redux-saga'

it('login renders correctly across screens', () => {
  const tree = render(<App />).toJSON();
  expect(tree).toMatchSnapshot()
});

test('login-test', () => {
  const {getByPlaceholderText, getByText, getAllByText} = render(
    <App />,
  );

  fireEvent.changeText(
    getByPlaceholderText('Enter UserName'),
    'Admin',
  );
  fireEvent.changeText(
    getByPlaceholderText('Enter password'),
    'Admin',
  );
  fireEvent.press(getByText('LOGIN'));

  const success = getAllByText('Login successful.');
  expect(success).toHaveLength(1); // expect 'success' message
});

test('login-test-fail', () => {
  const {getByPlaceholderText, getByText, getAllByText} = render(
    <App />,
  );

  fireEvent.changeText(
    getByPlaceholderText('Enter UserName'),
    'ABD',
  );
  fireEvent.changeText(
    getByPlaceholderText('Enter password'),
    '123',
  );
  fireEvent.press(getByText('LOGIN'));

  const fail = getAllByText('Login failed.');
  expect(fail).toHaveLength(1); // expect 'fail' message
});

test('get-posts', async() => {
  let dispatched = []
  const saga = runSaga(
    {
      dispatch: action => dispatched.push(action),
      getState: () => ({ state: 'test' }),
    },
    fetchPosts
  );
  const results = await saga.toPromise();

  expect(results.length).toBeGreaterThanOrEqual(1); // expect 'success' message
}, 10000);