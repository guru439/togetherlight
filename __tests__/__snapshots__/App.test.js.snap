// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders correctly across screens 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#fff",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    {
                      "color": "#0D094E",
                      "fontSize": 30,
                      "fontWeight": "bold",
                      "marginTop": 20,
                    }
                  }
                >
                  Welcome Back!
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          onGestureCancel={[Function]}
          pointerEvents="box-none"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        style={
                          {
                            "alignContent": "center",
                            "backgroundColor": "#fff",
                            "justifyContent": "center",
                          }
                        }
                      >
                        <RCTScrollView>
                          <View>
                            <View
                              style={
                                {
                                  "flexDirection": "row",
                                  "marginBottom": 40,
                                  "marginLeft": 25,
                                  "marginRight": 35,
                                  "marginTop": 10,
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "color": "#0D094E",
                                    "flex": 1,
                                    "fontSize": 15,
                                    "textAlign": "left",
                                  }
                                }
                              >
                                Hey! Good to see you again
                              </Text>
                            </View>
                            <View
                              style={
                                {
                                  "flexDirection": "row",
                                  "margin": 10,
                                  "marginLeft": 35,
                                  "marginRight": 35,
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "color": "#0D094E",
                                    "flex": 1,
                                    "fontSize": 30,
                                    "fontWeight": "bold",
                                    "textAlign": "left",
                                  }
                                }
                              >
                                Sign in
                              </Text>
                            </View>
                            <View
                              style={
                                {
                                  "backgroundColor": "#0D094E",
                                  "borderBottomLeftRadius": 20,
                                  "borderTopLeftRadius": 20,
                                  "borderTopRightRadius": 100,
                                  "margin": 10,
                                  "orderBottomRightRadius": 50,
                                }
                              }
                            >
                              <View
                                style={
                                  {
                                    "flexDirection": "row",
                                    "height": 60,
                                    "margin": 10,
                                    "marginLeft": 35,
                                    "marginRight": 35,
                                    "marginTop": 40,
                                  }
                                }
                              >
                                <TextInput
                                  autoCapitalize="none"
                                  blurOnSubmit={false}
                                  keyboardType="email-address"
                                  onChangeText={[Function]}
                                  onSubmitEditing={[Function]}
                                  placeholder="Enter UserName"
                                  placeholderTextColor="#8b9cb5"
                                  returnKeyType="next"
                                  style={
                                    {
                                      "borderBottomWidth": 2,
                                      "borderColor": "#ccc",
                                      "borderRadius": 10,
                                      "color": "#fff",
                                      "flex": 1,
                                      "paddingLeft": 15,
                                      "paddingRight": 15,
                                    }
                                  }
                                  underlineColorAndroid="#f000"
                                />
                              </View>
                              <View
                                style={
                                  {
                                    "flexDirection": "row",
                                    "height": 60,
                                    "margin": 10,
                                    "marginLeft": 35,
                                    "marginRight": 35,
                                    "marginTop": 40,
                                  }
                                }
                              >
                                <TextInput
                                  blurOnSubmit={false}
                                  keyboardType="default"
                                  onChangeText={[Function]}
                                  onSubmitEditing={[Function]}
                                  placeholder="Enter password"
                                  placeholderTextColor="#8b9cb5"
                                  returnKeyType="next"
                                  secureTextEntry={true}
                                  style={
                                    {
                                      "borderBottomWidth": 2,
                                      "borderColor": "#ccc",
                                      "borderRadius": 10,
                                      "color": "#fff",
                                      "flex": 1,
                                      "paddingLeft": 15,
                                      "paddingRight": 15,
                                    }
                                  }
                                  underlineColorAndroid="#f000"
                                />
                              </View>
                              <View
                                accessibilityState={
                                  {
                                    "busy": undefined,
                                    "checked": undefined,
                                    "disabled": undefined,
                                    "expanded": undefined,
                                    "selected": undefined,
                                  }
                                }
                                accessibilityValue={
                                  {
                                    "max": undefined,
                                    "min": undefined,
                                    "now": undefined,
                                    "text": undefined,
                                  }
                                }
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  {
                                    "alignItems": "center",
                                    "backgroundColor": "#fff",
                                    "borderBottomRightRadius": 40,
                                    "borderColor": "#7DE24E",
                                    "borderTopLeftRadius": 40,
                                    "borderWidth": 0,
                                    "color": "#0D094E",
                                    "height": 60,
                                    "justifyContent": "center",
                                    "marginBottom": 25,
                                    "marginLeft": 35,
                                    "marginRight": 35,
                                    "marginTop": 40,
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  onPress={[Function]}
                                  style={
                                    {
                                      "color": "#0D094E",
                                      "fontSize": 20,
                                      "fontWeight": "bold",
                                      "paddingVertical": 10,
                                    }
                                  }
                                >
                                  LOGIN
                                </Text>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
